x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/privileged_user_monitoring_onboarding/components/sample_dashboard/esql_data_generation.ts

```diff
@@ -56,10 +56,9 @@ export const generateListESQLQuery =
   };
 
 export const generateVisualizationESQLQuery =
-  (esqlSource: EsqlQueryOrInvalidFields) => (stackByField: string) => {
-    return map<string, string>(
+  (esqlSource: EsqlQueryOrInvalidFields) =>
+  (stackByField: string, timerange: { from: string; to: string }) =>
+    map<string, string>(
       (src) => `${src}
-        | EVAL timestamp=DATE_TRUNC(1 hour, TO_DATETIME(@timestamp))
-        | STATS results = COUNT(*) by timestamp, ${stackByField}`
+        | STATS results = COUNT(*) by timestamp = BUCKET(@timestamp, 30, "${timerange.from}", "${timerange.to}"), ${stackByField}`
```

Comment:
I think 30 should be assigned to a BUCKET_COUNT const with a comment saying that this jnumber just looks good so people aren't scared to change it in the future